### PR TITLE
openscenegraph: Opengl version selection

### PR DESCRIPTION
**Pull request by Makamitsu**
Specify library name and version:  openscenegraph/3.6.5

Relates to this issue : 
https://github.com/conan-io/conan-center-index/issues/12177

Basically, it just export opengl_version var from the cmake to the recipe as an conan option.

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/openscenegraph/all/conanfile.py [hunk 1]
```diff
@@ -41,6 +41,7 @@ class OpenSceneGraphConanFile(ConanFile):
         "with_png": [True, False],
         "with_tiff": [True, False],
         "with_zlib": [True, False],
+        "opengl_profile": ["GL1", "GL2", "GL3", "GLCORE", "GLES1", "GLES2", "GLES3", "GLES2+GLES3"],
```
Comment (uilianries) [inline recipes/openscenegraph/all/conanfile.py:44]: Use lower-case as preferred to keep the pattern, so you can use string.toupper() when passing to the cmake definition.

File: recipes/openscenegraph/all/conanfile.py [hunk 2]
```diff
@@ -68,6 +69,7 @@ class OpenSceneGraphConanFile(ConanFile):
         "with_png": True,
         "with_tiff": True,
         "with_zlib": True,
+        "opengl_profile": "GL2",
```
Comment (uilianries) [inline recipes/openscenegraph/all/conanfile.py:72]: Is it the default value on the upstream?

**Review by uilianries — COMMENTED** [review]
Thank you for your contribution, please, accept the CLA and also ask for access on the issue #4. Otherwise, the CI bot will not run/merger your pull request.

File: recipes/openscenegraph/all/conanfile.py (hunk 2 above)
Comment (Makamitsu) [reply recipes/openscenegraph/all/conanfile.py]: Yes, that's it. You can have a look exactly here, GL2 is assumed by default:
https://github.com/openscenegraph/OpenSceneGraph/blob/master/CMakeLists.txt#L491

Comment (uilianries) [reply recipes/openscenegraph/all/conanfile.py]: Thank you for checking!!

**Comment by uilianries** [comment]
@Makamitsu please, remove the pdb files from the package folder. They are not allowed in CCI packages.

File: recipes/openscenegraph/all/conanfile.py [hunk 3]
```diff
@@ -327,7 +337,7 @@ def setup_library(lib):
         if self.options.enable_windowing_system:
             if self.settings.os == "Linux":
                 library.requires.append("xorg::xorg")
-            elif tools.is_apple_os(self.settings.os):
+            elif is_apple_os(self.settings.os):
```
Comment (uilianries) [inline recipes/openscenegraph/all/conanfile.py:340]:
```suggestion
            elif is_apple_os(self):
```

File: recipes/openscenegraph/all/conanfile.py [hunk 4]
```diff
@@ -473,16 +483,16 @@ def setup_library(lib):
         # with_directshow
         # setup_plugin("directshow")
 
-        if tools.is_apple_os(self.settings.os):
+        if is_apple_os(self.settings.os):
```
Comment (uilianries) [inline recipes/openscenegraph/all/conanfile.py:486]:
```suggestion
        if is_apple_os(self):
```

**Pull request closed without merging**